### ISSUE TITLE
Visa Cal scraper stopped working

### ISSUE THREAD

**Issue by ifeins** [eshaham/israeli-bank-scrapers#200]
Recently the Visa Cal scraper stopped working.
When I've looked into it I see that it fails to login with the following error response returned:

```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN""http://www.w3.org/TR/html4/strict.dtd">
<HTML><HEAD><TITLE>Service Unavailable</TITLE>
<META HTTP-EQUIV="Content-Type" Content="text/html; charset=us-ascii"></HEAD>
<BODY><h2>Service Unavailable</h2>
<hr><p>HTTP Error 503. The service is unavailable.</p>
</BODY></HTML>
```

**Comment by sagiba** [comment]
Looks like some of the endpoints have changed, managed to tinker around and get it to work with the following changes:
1. Change BASE_URL to 'https://cal4u.cal-online.co.il/Cal4U'
2. Change authUrl to 'https://connect.cal-online.co.il/api/authentication/login'
3. Add the following header to every request: X-Site-Id: 8D37DF16-5812-4ACD-BAE7-CD1A5BFA2206
4. Change the auth header prefix from CalAuthScheme to CALAuthScheme (not sure this is necessary)
5. Add rememberMe: null to the auth request (not sure this is necessary)
6. Remove the check for Response.Status.Succeeded in auth response

I am working on a fork of the project with many changes so it's hard for me to submit a clean pull request, but I will try to get that later.

**Comment by ifeins** [comment]
@sagiba Wow, thanks for the quick reply.
Looking forward for the PR, let me know if I can help with anything.

**Comment by eshaham** [comment]
@sagiba looking forward to seeing that PR 😄

**Comment by ifeins** [comment]
@sagiba I've opened a PR with the changes you suggested, unfortunately the scraper still fails.
Can you let me know what should I fix?

https://github.com/eshaham/israeli-bank-scrapers/pull/201

**Comment by sagiba** [comment]
I created a new PR #202 based on #201 plus a few other changes that were necessary. It now works for me.

(didn't know if I can edit your existing PR)

**Comment by ifeins** [comment]
@sagiba I'm still unable to run the scraper using the playground.
I'm able to login to Visa Cal site using the username and password but the same credentials don't work in the scraper (I set them up using `npm run setup` and then `setup credentials`).

This is the error I'm getting:

```
  israeli-bank-scrapers git:(master) npm start

> israeli-bank-scrapers@0.6.7 start /Users/ifeins/Development/israeli-bank-scrapers
> node -r babel-register playground/scrape.js

Visa Cal: START_SCRAPING
Visa Cal: INITIALIZING
Visa Cal: LOGGING_IN
Visa Cal: TERMINATING
Visa Cal: END_SCRAPING
success: false
error type: INVALID_PASSWORD
error: undefined
```

Any idea what could be the issue?

**Comment by esakal** [comment]
Hi @ifeins ,
I reviewed the code again and `INVALID_PASSWORD` seems to be mapped only to that specific scenario. 

1. This scraper is using their mobile API if I understood correctly. If you can download their mobile app and make sure you are not blocked on that channel.
2. You can debug the playground using your IDE. This is the setup I did for webstorm to be able to debug the playground
![image](https://user-images.githubusercontent.com/4751797/58758352-007c3380-8523-11e9-9ede-e05753a003d0.png)
You will then be able to set breakpoint in `fetch.js` line 36 and check the request/response object complete graph.

Eran.

**Comment by sagiba** [comment]
I verified that it still works for me with a clean copy of eshaham/master.

Like @esakal suggested, please use the mobile app and see if it works for you. If it does, then you will have to sniff the app traffic to understand what it does differently in your scenario (I use Charles for that, can provide more detailed instructions if we get there). I cannot help because I cannot reproduce the issue.

**Comment by ifeins** [comment]
My account was blocked a second time, and now after they reset it, the scraper started working again :)
